src/js/module.js

```diff
@@ -8,3 +8,43 @@ Module.noImageDecoding = true;
 Module.noAudioDecoding = true;
 Module.noWasmDecoding = false; // we preload wasm using the built in plugin now
 Module.preloadedWasm = {};
+
+/**
+ *
+ * @param {undefined|(() => string)} stdin
+ * @param {undefined|((text: string) => void)} stdout
+ * @param {undefined|((text: string) => void)} stderr
+ */
+export function setStandardStreams(stdin, stdout, stderr) {
+  if(stdin) {
+    let input = [];
+    let inputIndex = -1; // -1 means that we just returned null
+    function stdinWrapper() {
+      if (inputIndex === -1) {
+        input = Module.intArrayFromString((stdin() || "") + "\n", true, 0);
```

Comment:
In any case it would be better to use `TextEncoder` here than `intArrayFromString`.
Also, what about the case where `stin()` returns `undefined` or `null`? Do we really want that to be the same as if it returned an empty string?

Reply:
Okay, I agree. It's probably better to use a proper web standard instead of emscripten internals.

I'm not sure how to best handle the case where `stdin` returns `undefined` or `null`. It would correspond to someone hitting the `cancel` button on a prompt...

Reply:
I guess our options are (1) throw or (2) return `null`?

Reply:
Okay, I took another look at this

Code
```py
print(input())
```

Input: Empty string (just hit ok on the prompt)
Output: `""` (also an empty string)

Input: `null` (hit cancel on the prompt)
Output: 
```
PythonError: Traceback (most recent call last):
  File "/lib/python3.8/asyncio/futures.py", line 178, in result
    raise self._exception
  File "/lib/python3.8/asyncio/tasks.py", line 280, in __step
    result = coro.send(None)
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 419, in eval_code_async
    return await CodeRunner(
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 276, in run_async
    res = eval(last_expr, self.globals, self.locals)
  File "<exec>", line 1, in <module>
EOFError: EOF when reading a line
 ```

Reply:
I guess it's your call as to whether you'd prefer an input of `null` (and also `undefined) to result in an empty user input or an exception.

Reply:
I think cancel is a bit like `Ctrl+C`. In the native console, that would raise `KeyboardInterrupt`. So I think raising `EOF` is good in that case.

Reply:
Sounds good then!